java/com/google/location/lbs/activity/audioset/dataprocessing/CsvParser.java

```diff
@@ -0,0 +1,244 @@
+   */
+  public CsvParser(String pathname) {
+    file = new File(pathname);
+    if (!file.exists()) {
+      throw new IllegalArgumentException(String.format("This file does not exist: %s", pathname));
+    }
+    ytidList = new ArrayList<>();
+    idToProtoMap = new HashMap<>();
+    idToUrlMap = new HashMap<>();
+  }
+
+  /** Tester Function to print 5 YouTube Links */
+  public void printYouTubeLinks() {
+    // prints youtube links
+    Scanner scan = null;
+    try {
+      scan = new Scanner(file);
+      String line = scan.nextLine();
+      while (line.charAt(0) == '#') {
+        line = scan.nextLine();
+      }
+      // while (scan.hasNext()) {
+      //   String ytid = line.split(",")[0];
+      //   String url = "https://youtube.com/watch?v=" + ytid;
+      //   System.out.println(url);
+      //   line = scan.nextLine();
+      // }
+      for (int i = 0; i < 5; i++) {
+        String ytid = line.split(",")[0];
+        String url = "https://youtube.com/watch?v=" + ytid;
+        System.out.println(url);
+        line = scan.nextLine();
+      }
+    } catch (IOException io) {
+      io.printStackTrace();
+    } finally {
+      if (scan != null) {
+        scan.close();
+      }
+    }
+  }
+
+  /** Extracts metadata from dataset.csv file */
+  public void csvToProtoMap() {
+    if (debug) System.out.println("in the csvToProtoMap method");
+    Scanner scan = null;
+    try {
+      scan = new Scanner(file);
+      int count = 0;
+      String line = scan.nextLine();
+      count++;
+      while (line.charAt(0) == '#') {
+        line = scan.nextLine();
+        count++;
+      }
+      do {
+        createProto(line);
+        if (debug) System.out.printf("At line %d\n", count);
+        line = scan.nextLine();
+        count++;
+      } while (scan.hasNext());
+      createProto(line);
+      if (debug) System.out.printf("At line %d\n", count);
+    } catch (IOException io) {
+      io.printStackTrace();
+    } finally {
+      if (scan != null) {
+        scan.close();
+      }
+    }
+  }
+
+  /**
+   * Creates a protocol buffer from the metadata in each line of the dataset.csv
+   *
+   * @param csvLine
+   */
+  public void createProto(String csvLine) {
+    if (debug) System.out.println("in the createProto method");
+    String[] lineArr = csvLine.split(",");
+    String ytid = lineArr[0];
+    ytidList.add(ytid);
+    String url = "https://youtube.com/watch?v=" + ytid;
+    idToUrlMap.put(ytid, url);
+    int start = (int) Double.parseDouble(lineArr[1]);
+    int end = (int) Double.parseDouble(lineArr[2]);
+    String[] labelsList = Arrays.copyOfRange(lineArr, 3, lineArr.length);
+    ArrayList<IncompleteExample.Example.Label> protoLabels = new ArrayList<>();
+    int len = labelsList.length;
+    for (int i = 0; i < len; i++) {
+      labelsList[i] = labelsList[i].replace("\"", "");
+      labelsList[i] = labelsList[i].trim();
+      IncompleteExample.Example.Label protoLabel =
```

Comment:
Is it possible to import `IncompleteExample.Example` instead so in code, we can simple use `Example`?

Reply:
I ran into some importing errors when I tried that the first time, but let me try again.